lib/auth/auth.go

```diff
@@ -313,27 +303,78 @@ func (s *AuthServer) GenerateServerKeys(hostID string, role teleport.Role) (*Pac
 	}, nil
 }
 
+// ValidteToken takes a provisioning token value and finds if it's valid. Returns
+// a list of roles this token allows its owner to assume, or an error if the token
+// cannot be found
+func (s *AuthServer) ValidateToken(token string) (roles teleport.Roles, e error) {
+	// look at static tokesn first:
+	for _, st := range s.StaticTokens {
+		if st.Value == token {
+			return st.Roles, nil
+		}
+	}
+	// look at the tokens in the token storage
+	tok, err := s.Provisioner.GetToken(token)
+	if err != nil {
+		log.Warn(err)
+		return nil, trace.Errorf("token not recognized")
+	}
+	return tok.Roles, nil
+}
+
+// enforceTokenTTL deletes the given token if it's TTL is over. Returns 'false'
+// if this token cannot be used
+func (s *AuthServer) checkTokenTTL(token string) bool {
```

Comment:
in general checkers in teleport return `error` if something goes wrong or `nil` if check has succeeded. This will spare you from logging errors everywhere, since you simply return them.